### ISSUE TITLE
Duplicate file guard is inconsistent between different OSes

### ISSUE THREAD

**Issue by igregson** [k-next/kirby#5178]
## Description

The duplicate file guard is inconsistent between different operating systems. 

**On case insensitive OSes (MacOS and Windows):** The guard prevents case insensitive duplicates.

**On case sensitive OSes (most Linux distros):** The guard _does not_ prevent case insensitive duplicates.

**Example**

On MacOS and Window, it's NOT possible to get both "dog.jpg" and "Dog.jpg"
On Linux, it IS possible to get both "dog.jpg" and "Dog.jpg"

**Expected behavior**  

Kirby's duplicate guard handling should work the same across all operating systems (as case insensitive, like on MacOS and Windows, since this prevents potential Git issues).

## My setup

**Kirby Version**  
3.9.4

**Comment by lukasbestle** [comment]
I agree, it makes sense to treat filenames case-insensitively in the duplicate checks (but only there). This would mean looking up existing files case-insensitively (e.g. by looking for the lowercase filename in a list of lowercase filenames) in the `FileRules` class.

**Comment by distantnative** [comment]
@lukasbestle looking at `FileRules::create()` this could also affect `File::exists()`

**Comment by lukasbestle** [comment]
Good question. Thinking about it, it will likely be very hard to fully emulate a case-insensitive filesystem. Especially here: Instead of calling `file_exists()`, we would have to get the full directory contents, lowercase all of them and check if the current file is in the list. A very expensive operation.

If we ignore Toolkit use cases and only look at files in the `content` folder, I wonder how far we actually need to go. If we prevent the creation or renaming of files when a file with the same name (ignoring case) already exists, there can never be a situation where two files with the same name exist. And since Kirby only operates on actually existing files it read from the content folder, we should then always be able to refer to each file by its actual filename, not by its potential names with different case.

That's why I wrote we only need to handle this in `FileRules` (= by not calling `$file->exists()` for the duplicate checks, but manually running the more expensive "simulated" case-insensitive filesystem algorithm). I might have missed something though.
